src/smhi/metobs.py

```diff
@@ -603,11 +605,18 @@ def _parse_data(self, raw_data: str) -> None:
             on_bad_lines="skip",
             usecols=[0, 1, 2, 3],
         )
+        columns = self.data.columns
+
+        if any([c for c in METOBS_PARAMETER_TIM if c in columns]):
+            datetime_columns = METOBS_PARAMETER_TIM
+        elif any([c for c in METOBS_PARAMETER_DYGN if c in columns]):
+            datetime_columns = METOBS_PARAMETER_DYGN
+        elif any([c for c in METOBS_PARAMETER_MANAD if c in columns]):
+            datetime_columns = METOBS_PARAMETER_MANAD
+        else:
+            raise TypeError("Can't parse type.")
```

Comment:
@docNord should be possible to be cast as a dict. Want to give it a try? Otherwise, I'm done with this as is for now.

Reply:
I wouldn't mind giving it a go tomorrow - I pulled the code yesterday and started looking, but I will gladly discard any changes I might have put in and look again tomorrow after the meeting.

Reply:
Let's keep it as is.